sympy/calculus/tests/test_util.py

```diff
@@ -63,6 +63,52 @@ def test_not_empty_in():
         Union(Interval(-2, -1, True, False), Interval(2, oo))
 
 
+def test_periodicity():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(sin(2*x), x) == pi
+    assert periodicity((-2)*tan(4*x), x) == pi/4
+    assert periodicity(sin(x)**2, x) == 2*pi
+    assert periodicity(3**tan(3*x), x) == pi/3
+    assert periodicity(tan(x)*cos(x), x) == 2*pi
+    assert periodicity(sin(x)**(tan(x)), x) == 2*pi
+    assert periodicity(tan(x)*sec(x), x) == 2*pi
+    assert periodicity(sin(2*x)*cos(2*x) - y, x) == pi/2
+    assert periodicity(tan(x) + cot(x), x) == pi
+    assert periodicity(sin(x) - cos(2*x), x) == 2*pi
+    assert periodicity(sin(x) - 1, x) == 2*pi
+    assert periodicity(sin(4*x) + sin(x)*cos(x), x) == pi
+    assert periodicity(exp(sin(x)), x) == 2*pi
+    assert periodicity(log(cot(2*x)) - sin(cos(2*x)), x) == pi
+    assert periodicity(sin(2*x)*exp(tan(x) - csc(2*x)), x) == pi
+    assert periodicity(cos(sec(x) - csc(2*x)), x) == 2*pi
+    assert periodicity(tan(sin(2*x)), x) == pi
+    raises(NotImplementedError, lambda: periodicity(exp(x), x))
+    raises(NotImplementedError, lambda: periodicity(log(x), x))
+    raises(NotImplementedError, lambda: periodicity(exp(x)**sin(x), x))
+
+
+def test_periodicity_check():
+    x = Symbol('x')
+    y = Symbol('y')
+
+    assert periodicity(tan(x), x, check=True) == pi
+    assert periodicity(sin(x) + cos(x), x, check=True) == 2*pi
```

Comment:
> I was thinking of what would happen, if we have a sum or a product with a constant argument. We should probably look at each argument separately.

Yes. That's exactly the basis of this implementation. :smile: 

> If there is constant arguments that return zero, that value could simply be omitted from the argument list of lcim.

This sounds good to me. It would definitely be easier to check for `0` rather than catching an exception.

Just to be clear, by `constant arguments` are you referring to actual constants (numbers) or all the arguments that are constant with respect to the given variable ?
For example:  `sin(y)` is independent of `x `. Should we return `0` here as well ?